analyzer/runtime/visitors.go

```diff
@@ -130,63 +130,69 @@ func VisitCall(call *sdkTypes.Call, result *sdkTypes.CallResult, handler *CallHa
 }
 
 type SdkEventHandler struct {
-	Core              func(event *core.Event) error
-	Accounts          func(event *accounts.Event) error
-	ConsensusAccounts func(event *consensusaccounts.Event) error
-	EVM               func(event *evm.Event) error
+	Core              func(event *core.Event, eventIdx int) error
+	Accounts          func(event *accounts.Event, eventIdx int) error
+	ConsensusAccounts func(event *consensusaccounts.Event, eventIdx int) error
+	EVM               func(event *evm.Event, eventIdx int) error
 }
 
-func VisitSdkEvent(event *nodeapi.RuntimeEvent, handler *SdkEventHandler) error {
+func VisitSdkEvent(event *nodeapi.RuntimeEvent, handler *SdkEventHandler, currIdx int) (int, error) {
 	if handler.Core != nil {
 		coreEvents, err := DecodeCoreEvent(event)
 		if err != nil {
-			return fmt.Errorf("decode core: %w", err)
+			return currIdx, fmt.Errorf("decode core: %w", err)
 		}
 		for i := range coreEvents {
-			if err = handler.Core(&coreEvents[i]); err != nil {
-				return fmt.Errorf("decoded event %d core: %w", i, err)
+			if err = handler.Core(&coreEvents[i], currIdx); err != nil {
+				return currIdx, fmt.Errorf("decoded event %d core: %w", i, err)
 			}
+			currIdx++
 		}
 	}
 	if handler.Accounts != nil {
 		accountEvents, err := DecodeAccountsEvent(event)
 		if err != nil {
-			return fmt.Errorf("decode accounts: %w", err)
+			return currIdx, fmt.Errorf("decode accounts: %w", err)
 		}
 		for i := range accountEvents {
-			if err = handler.Accounts(&accountEvents[i]); err != nil {
-				return fmt.Errorf("decoded event %d accounts: %w", i, err)
+			if err = handler.Accounts(&accountEvents[i], currIdx); err != nil {
+				return currIdx, fmt.Errorf("decoded event %d accounts: %w", i, err)
 			}
+			currIdx++
 		}
 	}
 	if handler.ConsensusAccounts != nil {
 		consensusAccountsEvents, err := DecodeConsensusAccountsEvent(event)
 		if err != nil {
-			return fmt.Errorf("decode consensus accounts: %w", err)
+			return currIdx, fmt.Errorf("decode consensus accounts: %w", err)
 		}
 		for i := range consensusAccountsEvents {
-			if err = handler.ConsensusAccounts(&consensusAccountsEvents[i]); err != nil {
-				return fmt.Errorf("decoded event %d consensus accounts: %w", i, err)
+			if err = handler.ConsensusAccounts(&consensusAccountsEvents[i], currIdx); err != nil {
+				return currIdx, fmt.Errorf("decoded event %d consensus accounts: %w", i, err)
 			}
+			currIdx++
 		}
 	}
 	if handler.EVM != nil {
 		evmEvents, err := DecodeEVMEvent(event)
 		if err != nil {
-			return fmt.Errorf("decode evm: %w", err)
+			return currIdx, fmt.Errorf("decode evm: %w", err)
 		}
 		for i := range evmEvents {
-			if err = handler.EVM(&evmEvents[i]); err != nil {
-				return fmt.Errorf("decoded event %d evm: %w", i, err)
+			if err = handler.EVM(&evmEvents[i], currIdx); err != nil {
+				return currIdx, fmt.Errorf("decoded event %d evm: %w", i, err)
 			}
+			currIdx++
 		}
 	}
-	return nil
+	return currIdx, nil
```

Comment:
This doesn't work ok with events that are currently not tracked and would be tracked in future. E.g. adding a handler for new event types would change the resulting indexes on reindex. 

The index should probably be made out of 2 values (index within GetEventsRaw, and then the index within the "DecodedEvents").

Reply:
do we skip events?

Reply:
> do we skip events?

Yeah for example rofl events are currently skipped (added in https://github.com/oasisprotocol/nexus/pull/812)

> come to think of it, how do oasis-core and oasis-sdk architecturally represent events? I think they're put into a map[event_type]slice_of_events kind of structure per block or tx

Yeah that's how it is done in oasis-sdk. So maybe going with (`event_type`, `index`(within the type)) pair would be the best, to match how oasis-sdk represents it?

Reply:
that sounds ok. I hope using a text column in the key is not too expensive